src/index.js

```diff
@@ -74,20 +73,39 @@ export default class Snuffles {
       requestOptions.body = JSON.stringify(casedBody)
     }
 
-    return fetch(`${url}${queryString}`, {
+    const urlWithQueryString = `${url}${queryString}`
+    this.log(urlWithQueryString, requestOptions)
+    return fetch(urlWithQueryString, {
       ...requestOptions
     })
       .then(res => {
         if (!res.ok) {
           const error = new Error('API response was not ok.')
+          this.log({...res, error})
           error.response = res
           throw error
         }
 
         return res
       })
-      .then(res => res.json())
-      .then(json => changeCaseObject.camelCase(json))
+      .then(res =>
+        res.json().then(json => {
+          const headers = {}
+          for (let pair of res.headers.entries()) {
```

Comment:
unfortunately this is quite a bit of extra effort just to have some additional information for logging, but maybe this could also be a useful option in the future for Snuffles to als return status & headers?

e.g. if you get a `201` response with a `location` header, this is something that Snuffles currently can't help you with.

Reply:
it could potentially be an alternative to make to log entries for responses:  one with headers and status, and a second one with the parsed body. But I don't really like that.

Reply:
concerning the idea of having the option that Snuffles can also return status & headers in addition to the body:  We could introduce an request option `fullResponse` which would be `false` by default and if you set it it to `true`, you don't get just the body, but an object `{ body, headers, status }` instead.  What do you think?

Reply:
This pretty much comes down to a discussion about what we want Snuffles to be I guess. Right now, it's more or less hit or miss, you either get a request body out of it you can use without any further modification, or it fails and you display the server error message or try again.

As this would only be an option, I see no problem in adding that, however, I'm not sure if there are more full fledged solution out there that are better suited for someone who would need to many information about the request?

Reply:
For now, I guess it would be fine just to keep it in mind as an idea.  We will see if we need it.

But we need to decide on whether we want to keep this code that parses the headers and status for the logging?

Reply:
I'd vote for keeping it! I guess for debugging it can be super helpful to have some more information about the request